core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -392,10 +397,36 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * Request the cluster manager to remove the given executor.
+   * Request the cluster manager to remove the given executors.
    * Return whether the request is received.
    */
-  private def removeExecutor(executorId: String): Boolean = synchronized {
+  private def removeExecutors(executorIds: Seq[String]): Boolean = synchronized {
+
+    val executorIdsToBeRemoved = executorIds.filter(canBeKilled)
+
+    // Send a request to the backend to kill this executor
+    val removeRequestAcknowledged = testing || client.killExecutors(executorIdsToBeRemoved)
```

Comment:
As @vanzin mentioned, the ```CoarseGrainedSchedulerBackend``` filters out executors based on whether we want to force kill the executors which are already running and the one's which are already pending to be removed. 

In our case, we are requesting to kill the executors which have remained idle for the configured timeout duration. AFAIK this does not lead to filtering out any executors unless they are already marked for removal which is harmless.